### DISCUSSION TITLE
How to create captcha verification using laravel?

### DISCUSSION THREAD

**Discussion by wayosu** [community/community#72817] [Programming Help]
### Body

I want to use captcha verification on the registration form.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by wayosu** [comment]
Creating a CAPTCHA verification system in Laravel can help protect your website or application from automated bots and spam. You can use the "laravel-captcha" package to easily integrate CAPTCHA into your Laravel project. Here's a step-by-step guide on how to set up CAPTCHA verification in Laravel using the "laravel-captcha" package:

1. Install the "laravel-captcha" Package:
Use Composer to install the "laravel-captcha" package.
`composer require mews/captcha`

2. Service Provider Configuration:
After installing the package, add the service provider to your **config/app.php** file in the **providers** array:
`'providers' => [
    // ...
    Mews\Captcha\CaptchaServiceProvider::class,
],`

3. Publish Configuration and Assets:
Publish the package's configuration file and assets (fonts and images) to your Laravel project:
`php artisan vendor:publish --tag=captcha`

4. Generate CAPTCHA Image:
In your controller or route handler, generate a CAPTCHA image and display it in a view. Here's an example route handler:
`use Mews\Captcha\Facades\Captcha;`
`public function showCaptcha()
{
    $captcha = Captcha::create();
    return view('captcha', compact('captcha'));
}`

5. Display CAPTCHA Image in a View:
Create a view (e.g., captcha.blade.php) to display the CAPTCHA image along with an input field for users to enter the CAPTCHA code.
`<img src="{{ $captcha['image'] }}" alt="CAPTCHA">`
`<input type="text" name="captcha">`

6. Validate CAPTCHA:
In your form processing logic, validate the CAPTCHA code provided by the user. You can use Laravel's built-in validation to do this:
`use Illuminate\Http\Request;`
`public function processForm(Request $request)
{
    $request->validate([
        'captcha' => 'required|captcha',
        // Add other validation rules here.
    ]);

    // If the CAPTCHA is valid, continue processing the form.
}`
The 'captcha' validation rule is provided by the "laravel-captcha" package.

7. Customize CAPTCHA Settings:
You can customize the CAPTCHA appearance and behavior by editing the configuration file located at config/captcha.php. For example, you can change font size, image size, and more.

8. Testing:
Test your CAPTCHA system by loading the CAPTCHA image, submitting the form, and verifying that the CAPTCHA code is validated correctly.

**Comment by mamoon-rasheed** [comment]
use "reCaptcha" by Google
create an account by visiting the reCaptcha by google website
generate API keys

install recaptcha package on your laravel app
`composer require anhskohbo/no-captcha`

configure laravel to use the package
`'providers' => [
    // ...
    Anhskohbo\NoCaptcha\NoCaptchaServiceProvider::class,
],`

also don't forget to add the alias
`
'aliases' => [
    // ...
    'NoCaptcha' => Anhskohbo\NoCaptcha\Facades\NoCaptcha::class,
],`

add the configuration for recaptcha into your .env file
NOCAPTCHA_SECRET=your-secret-key
NOCAPTCHA_SITEKEY=your-site-key


use captcha on your view
`
<form action="/your-form-action" method="POST">
    @csrf
    <label for="name">Name:</label>
    <input type="text" name="name" id="name" required>

    @captcha

    <button type="submit">Submit</button>
</form>
`
handle the captcha verification at your controller
`
use Illuminate\Http\Request;
use Illuminate\Support\Facades\Validator;
use Anhskohbo\NoCaptcha\NoCaptcha;

public function submitForm(Request $request, NoCaptcha $noCaptcha)
{
    // Validate the CAPTCHA
    $validator = Validator::make($request->all(), [
        'name' => 'required',
        'g-recaptcha-response' => 'required|captcha'
    ]);

    if ($validator->fails()) {
        // CAPTCHA validation failed
        return back()->withErrors($validator)->withInput();
    }

    // CAPTCHA validation passed, continue processing the form
    // ...

    return redirect()->to('/success-page');
}
`
you can also use it like this
`
<div class="g-recaptcha" data-sitekey="{{ env('NOCAPTCHA_SITEKEY') }}"></div>
`
`
use Illuminate\Http\Request;
use Illuminate\Support\Facades\Validator;

public function store(Request $request)
{
    // Validate the reCAPTCHA response
    $validator = Validator::make($request->all(), [
        'g-recaptcha-response' => 'required|captcha',
    ]);

    if ($validator->fails()) {
        // CAPTCHA validation failed, return an error response or redirect back with an error message.
        return back()->withErrors($validator)->withInput();
    }

    // CAPTCHA validation succeeded, continue with your form processing logic.
    // ...
}
`

**Reply by wayosu** [reply]
thank you brother

**Comment by coderMohammed1** [comment]
just watch this video:
https://www.youtube.com/watch?v=NsFSA-MQjeY

**Reply by wayosu** [reply]
thank u brooo
